gradle-baseline-java/src/main/groovy/com/palantir/baseline/extensions/BaselineJavaVersionsExtension.java

```diff
@@ -31,7 +33,8 @@ public class BaselineJavaVersionsExtension {
     private final Property<JavaLanguageVersion> libraryTarget;
     private final Property<JavaLanguageVersion> distributionTarget;
     private final Property<JavaLanguageVersion> runtime;
-    private final MapProperty<JavaLanguageVersion, JavaInstallationMetadata> jdks;
+    private final LazilyConfiguredMapping<JavaLanguageVersion, AtomicReference<JavaInstallationMetadata>> jdks =
```

Comment:
Why use an `AtomicReference` here? I wanted to make a general type I could [reuse in gradle-jdks](https://github.com/palantir/gradle-jdks/pull/19/files#r866173898) to configure the versions/distributions, which would involve modifying extension objects in the regular gradle style. So we use an `AtomicReference` here just to make it act like a box you can store data in.